### DISCUSSION TITLE
Nginx Proxy manager - default.json

### DISCUSSION THREAD

**Discussion by danmed** [tteck/Proxmox#205] [General]
Hi,

I'm looking at switching from a docker container to your LXC for NPM but i use an external MariaDB database.

I've found the default.json where i set the connection string but it's in a folder with the NPM version number : 

/nginx-proxy-manager-2.9.18/backend/config/default.json

I am concerned that when i upgrade the lxc using your update script, any changes i make to this file will be lost.. 

Am i right to be concerned or does this file get moved to the new version too?

Edit : OK, now i'm looking at that file, it's already set to mysql.. which says to me it's not being used as i see no MySQL or MariaDB processes running in the LXC.... so.. i guess my question is.. how do i point this LXC to an external MariaDB server?

**Comment by tteck** [comment]
This installation is using sqlite3 _only_, contained in the LXC (`/data/database.sqlite`) so that the app and all data can be backed up and restored together.

**Comment by solidus1983** [comment]
Sorry to bump an old thread however i have the same question and i have been trying to get it to use the external database i currently use for my docker version.

Normally the docker version parse the database information during the service booting which has worked grand.

However looking at your script you do create a `production.json` which has

```
{
  "database": {
    "engine": "knex-native",
    "knex": {
      "client": "sqlite3",
      "connection": {
        "filename": "/data/database.sqlite"
      }
    }
  }
}
```


so using the very same file but altering it for mysql/maria like
```
{
  "database": {
    "engine": "mysql",
    "host": "db",
    "name": "npm",
    "user": "npm",
    "password": "npm",
    "port": 3306
      }
    }
  }
}
```
I have changed the usr and pwd to that of the default.json file.

However it still makes a `/data/database.sqlite` file and i can't login using the creds that are stored in the remote database.

**Reply by tteck** [reply]
The extra closing curly braces in your code will cause a syntax error.
https://github.com/NginxProxyManager/nginx-proxy-manager/blob/develop/backend/config/default.json
